src/launcher/launcher.cxx

```
@@ -109,13 +124,52 @@ void CreateGuiltyGearProcess(
 	ResumeThread(pi.hThread);
 }
 
+std::string GetLatestReleasePublicationDate() {
+	std::wstring answer = GetHttpsRequest(GITHUB_API_BASE_URI, GITHUB_API_LATEST_RELEASE_ENDPOINT, HTTP_USER_AGENT);
+	json resJson = json::parse(answer, NULL, false);
+	if (resJson.contains("published_at")) { 
+		return resJson["published_at"];
+	}
+	return "";
+}
+
+bool Update() {
+	std::string latestReleasePublicationDate = GetLatestReleasePublicationDate();
+
+	if (latestReleasePublicationDate == "") {
+		return false;
+	}
+
+	if (ParseISO8601(_BUILD_DATETIME_ISO8601_) < ParseISO8601(latestReleasePublicationDate))
+	{
+		const int result = MessageBox(NULL, NEW_VERSION_AVAILABLE_MESSAGE.c_str(), NEW_VERSION_AVAILABLE_TITLE.c_str(), MB_YESNO);
+
+		switch (result)
+		{
+		case IDYES:
+			ShellExecuteA(NULL, "open", LATEST_RELEASE_URI.c_str(), NULL, NULL, SW_SHOWNORMAL);
+			return true;
+			break;
+		case IDNO:
+			return false;
+			break;
+		}
+	}
+
+	return false;
+}
+
 int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE, PWSTR pCmdLine, int nCmdShow)
 {
 	wchar_t szGuiltyDirectory[1024] = { 0 };
 	wchar_t szGuiltyExePath[1024] = { 0 };
 	char szSidecarConfigPathA[1024] = { 0 };
 	char szSidecarDllPathA[1024] = { 0 };
 
+	if (Update()) {
```

What's the effect on the user if they're not online when they run this? Or are on a slow connection?

---

Not online: Instantly fails silently. Launcher.exe continues and launches the game.
Slow connection: Tested with https://jagt.github.io/clumsy/, incredibly disappointing. Delays game launch by several seconds. After research it seems the timeout flags on the `WinInet` interface are not functional at all. I'll experiment with `WinHTTP` instead.

---

The timeout flags in winhttp weren't very useful either so I went with the following in https://github.com/FIXMESTEVE/GGPOPLUSR/commit/1fe4e2b183aa4d17fd1dba831cffea4671863de8:

The request is run in a thread that Launcher.exe will await for a maximum amount of `timeout` milliseconds. When the thread finishes Launcher.exe retrieves its result and resumes immediately (so if the thread finishes in 10ms Launcher.exe resumes after those 10ms), or will kill the thread if `timeout` is reached and resume.
It passes the crap internet™️ test, meaning the game launch is not massively delayed if you are on superslow and faulty net.